packages/truffle-contract/lib/handlers.js

```diff
@@ -104,6 +104,7 @@ var handlers = {
     // .method(): resolve/reject receipt in handler
     if (parseInt(receipt.status) == 0 && !context.onlyEmitReceipt){
       var error = new StatusError(context.params, receipt.transactionHash, receipt);
+      error.reason = context.reason;
```

Comment:
I'm a little bit worried about attaching `reason` to these error objects like this. It feels like an anti-pattern, but maybe I'm just afraid of statefulness. Let's keep an eye on this in the beta, in case it causes unforeseen complications.

Pardon me if I haven't seen it yet, but it might be worth making a special note somewhere in a comment, so we can encourage future work to be mindful of pre- vs. post-reason-attach errors. I know you've written a good matrix of tests around this, so it's probably safe enough, esp. since it's not likely to go without `undefined` checks anyway.

Reply:
I've moved this into the StatusError file where it (possibly) looks less arbitrary. Truffle-contract has actually attached quite a bit of stuff to the error traditionally - like the txHash for example (which we continue to do for backwards compatibility).